modules/core/include/opencv2/core/hal/intrin_neon.hpp

```diff
@@ -2228,9 +2228,7 @@ inline v_int16x8 v_round(const v_float16x8 &a)
 
 inline v_int16x8 v_floor(const v_float16x8 &a)
 {
-    int16x8_t a1 = vcvtq_s16_f16(a.val);
-    uint16x8_t mask = vcgtq_f16(vcvtq_f16_s16(a1), a.val);
-    return v_int16x8(vaddq_s16(a1, vreinterpretq_s16_u16(mask)));
+    return v_int16x8(vcvtmq_s16_f16(a.val));
```

Comment:
The instruction is not available on armv7, but available on aarch32 and aarch64. We need macro check here.

Reply:
Do we have an existing macro indicating the target platform is armv7?

Reply:
Input is of type `v_float16x8`. I guess compile flag `+fp16` can only be used with >= `armv8.2a` right? In this case, no need to guard with `__ARM_ARCH > 7` then.